### PR TITLE
TDI-38097:Create conversion for most used functions

### PR DESCRIPTION
**Pull request by pyzhou-talend**
https://jira.talendforge.org/browse/TDI-38097

### CODE REVIEW

File: main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+			throw new IllegalArgumentException("Parameter searchAndResult should be in pair.");
+		}
+		Map<T, Y> search = new HashMap<T, Y>();
+		for (int i = 0; i < searchAndResult.length; i += 2) {
+			search.put(searchAndResult[i], (Y) searchAndResult[i + 1]);
+		}
+		return DECODE(value, defaultValue, search);
+	}
+
+	public String DECODE(String value, String defaultValue, String... searchAndResult) {
```
Comment (pteyssier) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java]: Duplicate code

**Comment by pyzhou-talend** [comment]
As tested, only ISNULL working for tmap now.

File: main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+     * @param start 
+     * 		The default is 1, meaning that INSTR starts the search at the first character in the string.
+     * @param occurrence
+     * 		If the search value appears more than once in the string, you can specify which occurrence you want to search for.
+     * @param comparison_type
+     * 		default 0: INSTR performs a linguistic string comparison. 1: INSTR performs a binary string comparison.
+     * @return
+     * {example} new StringHandling<String>.INSTR("This is a test","t",1,2,0) #14
+     */
+	public Integer INSTR(T string, T search_value, Integer start, Integer occurrence, Integer comparison_type) {
```
Comment (pteyssier) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java]: Not necessary to put it here using generic type. Better to define 2 method with String and Byte array

File: main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+				throw new RuntimeException("Please enter a vaild 'search_value' or 'binary search_value'.");
+			}
+		}
+
+		if (start != null && start != 0) {
+			defultStart = start;
+		}
+		if (occurrence != null) {
+			if (occurrence <= 0) {
+				throw new RuntimeException(
```
Comment (pteyssier) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/TalendStringUtil.java]: use illegalargumentexception

File: main/plugins/org.talend.librariesmanager/resources/java/routines/StringHandling.java [hunk 4]
```diff
@@ -542,5 +542,174 @@ public String RPAD(String first_string, int length, String second_string) {
...
+     * 		If the search value appears more than once in the string, you can specify which occurrence you want to search for.
+     * @param comparison_type
+     * 		default 0: INSTR performs a linguistic string comparison. 1: INSTR performs a binary string comparison.
+     * @return
+     * {example} new StringHandling<String>.INSTR("This is a test","t",1,2,0) #14
+     */
+	
+    public Integer INSTR(String string, String search_value, Integer start, Integer occurrence) {
+
+        int defultStart = 1;
```
Comment (pteyssier) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/StringHandling.java]: typo default

**Pull request merged**